sshd-core/src/main/java/org/apache/sshd/client/session/ClientSession.java

```diff
@@ -304,10 +311,10 @@ default void executeRemoteCommand(
              ClientChannel channel = createExecChannel(command, charset, null, Collections.emptyMap())) {
             channel.setOut(channelOut);
             channel.setErr(channelErr);
-            channel.open().await(); // TODO use verify and a configurable timeout
 
-            // TODO use a configurable timeout
-            Collection<ClientChannelEvent> waitMask = channel.waitFor(REMOTE_COMMAND_WAIT_EVENTS, 0L);
+            channel.open().verify(commandTimeoutMillis, TimeUnit.MICROSECONDS);
+
+            Collection<ClientChannelEvent> waitMask = channel.waitFor(REMOTE_COMMAND_WAIT_EVENTS, commandTimeoutMillis);
```

Comment:
Here we should probably wait only for commandTimeoutMillis minus the time we spent waiting for the channel to open.